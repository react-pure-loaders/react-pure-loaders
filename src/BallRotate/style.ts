import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';

const rotate = keyframes({
  '0%': {
    transform: 'rotate(0deg) scale(1)',
  },
  '100%': {
    transform: 'rotate(360deg) scale(1)',
  },
  '50%': {
    transform: 'rotate(180deg) scale(.6)',
  },
});

export default (color?: string) => ({
  position: 'relative' as 'relative',
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    position: 'relative' as 'relative',

    '&:first-of-type': {
      animation: `${rotate} 1s 0s cubic-bezier(.7,-.13,.22,.86) infinite`,

    },

    '&:before, &:after': {
      ...balls(color),
      content: '\'\'',
      opacity: 0.8,
      position: 'absolute' as 'absolute',
    },

    '&:before': {
      left: '-28px',
      top: '0px',
    },

    '&:after': {
      left: '25px',
      top: '0px',
    },
  },
});
