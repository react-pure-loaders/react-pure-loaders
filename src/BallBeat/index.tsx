/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
