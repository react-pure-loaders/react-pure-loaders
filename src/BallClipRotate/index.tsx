/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
