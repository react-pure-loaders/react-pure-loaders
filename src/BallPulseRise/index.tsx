/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
