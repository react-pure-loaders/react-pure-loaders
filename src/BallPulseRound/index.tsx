/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
