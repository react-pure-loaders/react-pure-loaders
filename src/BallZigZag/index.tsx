/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
