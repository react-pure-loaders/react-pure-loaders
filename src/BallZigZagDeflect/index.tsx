/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
