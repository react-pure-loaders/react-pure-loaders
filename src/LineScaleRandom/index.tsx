/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
