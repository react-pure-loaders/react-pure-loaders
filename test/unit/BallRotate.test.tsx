import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallRotate from '../../src/BallRotate';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallRotate>', () => {
    afterEach(cleanup);

    test('BallRotate should match snapshot', () => {
        const { container } = render(<BallRotate loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallRotate should have default color', () => {
        const { container } = render(<BallRotate loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('BallRotate should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallRotate color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallRotate should have no children', () => {
        const { container } = render(<BallRotate loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallRotate should have three children', () => {
        const { container } = render(<BallRotate loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(2);
    });
});
