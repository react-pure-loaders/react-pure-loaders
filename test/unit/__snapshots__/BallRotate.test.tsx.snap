// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallRotate> BallRotate should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg) scale(1);
    -moz-transform: rotate(0deg) scale(1);
    -ms-transform: rotate(0deg) scale(1);
    transform: rotate(0deg) scale(1);
  }

  100% {
    -webkit-transform: rotate(360deg) scale(1);
    -moz-transform: rotate(360deg) scale(1);
    -ms-transform: rotate(360deg) scale(1);
    transform: rotate(360deg) scale(1);
  }

  50% {
    -webkit-transform: rotate(180deg) scale(.6);
    -moz-transform: rotate(180deg) scale(.6);
    -ms-transform: rotate(180deg) scale(.6);
    transform: rotate(180deg) scale(.6);
  }
}

.emotion-0 {
  position: relative;
}

.emotion-0>div {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  position: relative;
}

.emotion-0>div:first-of-type {
  -webkit-animation: animation-0 1s 0s cubic-bezier(.7,-.13,.22,.86) infinite;
  animation: animation-0 1s 0s cubic-bezier(.7,-.13,.22,.86) infinite;
}

.emotion-0>div:before,
.emotion-0>div:after {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  content: '';
  opacity: 0.8;
  position: absolute;
}

.emotion-0>div:before {
  left: -28px;
  top: 0px;
}

.emotion-0>div:after {
  left: 25px;
  top: 0px;
}

<div
  class="emotion-0"
>
  <div />
</div>
`;
